src/components/Home/Home.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+
+import SearchField from '../SearchField';
+import SearchableDropdown from '../SearchableDropdown';
+import CurrentWeather from '../CurrentWeather/CurrentWeatherContainer';
+
+import useLocalStorage from '../CustomHooks/useLocalStorage';
+import { ENTER_KEYCODE, MAX_ITEMS_LENGTH } from '../../common/constants';
+import { groupQueryString, transformSpaces } from '../../common/helpers';
+import { refreshAccessToken } from '../../common/auth';
+import * as S from '../Theme/GlobalStyles';
+
+const Home = ({ className }) => {
+  const history = useHistory();
```

Comment:
a lot of stuff going on here. Hard to distinguish. I suggest to start creating abstactions like custom hooks here